src/components/sdk/MeetingControls/VideoInputBackgroundBlurControl.tsx

```diff
@@ -0,0 +1,172 @@
+import { useLocalVideo } from '../../../providers/LocalVideoProvider';
+import { useMeetingManager } from '../../../providers/MeetingProvider';
+import { DeviceConfig } from '../../../types';
+import {
+  isOptionActive,
+  videoInputSelectionToDevice,
+} from '../../../utils/device-utils';
+import { ControlBarButton } from '../../ui/ControlBar/ControlBarButton';
+import { Camera, Spinner } from '../../ui/icons';
+import PopOverItem from '../../ui/PopOver/PopOverItem';
+import PopOverSeparator from '../../ui/PopOver/PopOverSeparator';
+
+interface Props {
+  /** The label that will be shown for video input control, it defaults to `Video`. */
+  label?: string;
+  /** The label that will be shown when the Background Blur checkbox is checked */
+  backgroundBlurOnLabel?: string;
+  /** The label that will be shown when the Background Blur checkbox is not checked */
+  backgroundBlurOffLabel?: string;
+}
+
+const videoInputConfig: DeviceConfig = {
+  additionalDevices: true,
+};
+
+const VideoInputBackgroundBlurControl: React.FC<Props> = ({
+  label = 'Video',
+  backgroundBlurOnLabel = 'Background Blur Enabled',
+  backgroundBlurOffLabel = 'Enable Background Blur',
+}) => {
+  const meetingManager = useMeetingManager();
+  const { devices, selectedDevice } = useVideoInputs(videoInputConfig);
+  const { isVideoEnabled, toggleVideo } = useLocalVideo();
+  const { isBackgroundBlurSupported, createBackgroundBlurDevice } =
+    useBackgroundBlur();
+  const [isVideoTransformCheckBoxOn, setisVideoTransformCheckBoxOn] =
+    useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [
+    dropdownWithVideoTransformOptions,
+    setDropdownWithVideoTransformOptions,
+  ] = useState<ReactNode[] | null>(null);
+  const [activeVideoDevice, setDevice] = useState<
+    Device | VideoTransformDevice | null
+  >(meetingManager.selectedVideoInputDevice);
+
+  // TODO: Move this to the Video Input Provider and expose only one selected Video Input device state
+  useEffect(() => {
+    meetingManager.subscribeToSelectedVideoInputTransformDevice(setDevice);
+    return () => {
+      meetingManager.unsubscribeFromSelectedVideoInputTranformDevice(setDevice);
+    };
+  }, []);
+
+  useEffect(() => {
+    const deviceOptions: ReactNode[] = devices.map((option) => (
+      <PopOverItem
+        key={option.deviceId}
+        children={<span>{option.label}</span>}
+        checked={isOptionActive(selectedDevice, option.deviceId)}
+        onClick={async (): Promise<void> => {
+          // If background blur is on, then re-use the same video transform pipeline, but replace the inner device
+          // If background blur is not on, then do a normal video selection
+          if (isVideoTransformDevice(activeVideoDevice) && !isLoading) {
+            setIsLoading(true);
+            const receivedDevice = videoInputSelectionToDevice(option.deviceId);
+            if (activeVideoDevice instanceof DefaultVideoTransformDevice) {
```

Comment:
Do we still need to check this even after we already check isVideoTransformDevice ?

Reply:
Should we check if its VideoTransformDevice and implements the chooseNewInnerDevice method then? Richard did something similar when he added the isDestroyable and destroy methods in JS SDK. Cause, this is a hard-coded check and would be beneficial to have this check against and interface. There may be changes later in JS SDK or say if someone passes a different transform device object which still supports the inner device concept, then we would have to come and change this to support later. Any suggestion @ltrung ? (I guess can be done as a follow-up)
https://github.com/aws/amazon-chime-sdk-js/blob/master/src/destroyable/Destroyable.ts

Reply:
There actually isnt a `BackgroundBlurVideoTransformDevice`, only the `DefaultVideoTransformDevice`. How else could I check if the chooseNewInnerDevice is able to be called?